go/cmd/dolt/dolt.go

```diff
@@ -144,7 +145,11 @@ func runMain() int {
 
 	warnIfMaxFilesTooLow()
 
-	dEnv := env.Load(context.TODO(), env.GetCurrentUserHomeDir, filesys.LocalFS, doltdb.LocalDirDoltDB, Version)
+	dEnv := env.Load(context.TODO(), env.GetCurrentUserHomeDir, filesys.LocalFS, doltdb.LocalDirDoltDB, doltdb.DoltVersion)
+
+	if dEnv.DBLoadError == nil {
+		dEnv.DBLoadError = rebase.MaybeMigrateUniqueTags(context.Background(), dEnv)
+	}
```

Comment:
@bheni This where we potentially migrate repos if the repo was created with an older Dolt version.
If this function doensn't find the Dolt version written to the root at each of the branches, it will performa a migration.
Going forward we will write the most recent dolt version to the root value.